### PR TITLE
Fix threaded cache receipt when event holds multiple receipts

### PR DESCRIPTION
**Pull request by gsouquet**
First step for https://github.com/vector-im/element-web/issues/24164

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix threaded cache receipt when event holds multiple receipts ([\#3026](https://github.com/matrix-org/matrix-js-sdk/pull/3026)).

### CODE REVIEW

**Review by weeman1337 — CHANGES_REQUESTED** [review]
Checked out the branch locally. The issue still occurs with the changes :frowning_face:

**Comment by gsouquet** [comment]
I changed the description so that it does not close https://github.com/vector-im/element-web/issues/24164 . But this change is still valid.

There are more changes coming from https://github.com/matrix-org/matrix-js-sdk/pull/3020 that I'm going to add that will fix the issue you raised

File: src/models/thread.ts [hunk 1]
```diff
@@ -317,17 +317,9 @@ export class Thread extends ReadReceipt<EmittedEvents, EventHandlerMap> {
...
-                Object.keys(content[eventId]).forEach((receiptType: ReceiptType | string) => {
-                    Object.keys(content[eventId][receiptType]).forEach((userId: string) => {
-                        const receipt = content[eventId][receiptType][userId] as Receipt;
-                        this.addReceiptToStructure(eventId, receiptType as ReceiptType, userId, receipt, synthetic);
-                    });
-                });
-            });
+    private processReceipts(receipts: CachedReceiptStructure[] = []): void {
+        for (const { eventId, receiptType, userId, receipt, synthetic } of receipts) {
+            this.addReceiptToStructure(eventId, receiptType as ReceiptType, userId, receipt, synthetic);
```
Comment (weeman1337) [inline src/models/thread.ts:322]: Do we need the cast here? If it can be a string, shouldn't `addReceiptToStructure` also accept a string?

**Review by weeman1337 — APPROVED** [review]
LGTM :+1: One smaller comment

**Pull request merged**